src/Compilers/CSharp/Test/Semantic/Semantics/LambdaTests.cs

```diff
@@ -4345,15 +4732,27 @@ static void Main()
                 obj = null;
                 return true;
             };
-    }
+
+        D2 d2 = ([NotNullWhen(true)] out object? obj) =>
+            {
+                obj = null;
+                return false;
+            };
+     }
 }";
             var comp = CreateCompilation(new[] { source, NotNullWhenAttributeDefinition }, parseOptions: TestOptions.RegularPreview);
-            // https://github.com/dotnet/roslyn/issues/52827: Should report WRN_ParameterConditionallyDisallowsNull.
-            comp.VerifyDiagnostics();
+            comp.VerifyDiagnostics(
+                // (10,15): warning CS8622: Nullability of reference types in type of parameter 'obj' of 'lambda expression' doesn't match the target delegate 'D' (possibly because of nullability attributes).
+                //         D d = ([NotNullWhen(true)] out object? obj) =>
+                Diagnostic(ErrorCode.WRN_NullabilityMismatchInParameterTypeOfTargetDelegate, "([NotNullWhen(true)] out object? obj) =>").WithArguments("obj", "lambda expression", "D").WithLocation(10, 15),
+                // (13,17): warning CS8762: Parameter 'obj' must have a non-null value when exiting with 'true'.
+                //                 return true;
+                Diagnostic(ErrorCode.WRN_ParameterConditionallyDisallowsNull, "return true;").WithArguments("obj", "true").WithLocation(13, 17)
```

Comment:
>ErrorCode.WRN_ParameterConditionallyDisallowsNull [](http://example.com/codeflow?start=27&length=49)

Should this be reported for the previous lambda as well?

Reply:
One lambda returns true with obj=null and the other returns false. So only one of them reports this warning.